client/src/main/webjar/map/map.component.js

```diff
@@ -16,6 +16,31 @@
         var map = this;
 
         /**
+         * Returns the current time in milliseconds so we can bust google's caching of our generated KML files.
+         * @returns {Number} current time in milliseconds.
+         */
+        function getMsTime() {
+            return Date.now();
+        }
+
+        /**
+         * Returns the program IDs that the user is interested in as a comma delimited string.
+         * @returns {String} comma delimited string of the IDS of the programs to display.
+         */
+        function getProgramIdsString() {
+            return "1,2";
+        }
+
+        /**
+         * Returns the url for google to get a generated KML file.
+         * @returns {String} full URL to generate a KML file.
+         */
+        function getKmlUrl() {
+            return "https://hmhb-ryan.herokuapp.com/api/kml?bc=" + getMsTime() +
```

Comment:
I need to scavenge a piece of my authentication work (I created something that would reconstruct the protocol+host+port, etc.).